docs/codeql/codeql-cli/about-codeql-workspaces.rst

```diff
@@ -0,0 +1,94 @@
+
+
+CodeQL workspaces and query resolution
+--------------------------------------
+
+All CodeQL packs in a workspace are available as source dependencies for each other when you run any CodeQL command that resolves queries or packs. For example, when you run ``codeql pack install`` in a pack directory in a workspace, any dependency that can be found in the workspace will be instead of downloading that dependency to the package cache and adding it to the ``codeql-pack.lock.yml`` file. For more information, see `:ref:Adding and Installing Dependencies <adding-and-installing-dependencies>`__.
+
+Similarly, when you publish a CodeQL query pack to the GitHub container registry using  ``codeql pack publish`` the command will always use the dependencies from the workspace instead of using dependencies found in the local package cache.
+
+This ensures that any local changes you make to a query library in a dependency are automatically reflected in any query packs you publish from that workspace.
+
+.. pull-quote::
+
+  Note
+
+  Source dependencies are CodeQL packs that are resolved from the local file system. These dependencies can be in the same CodeQL workspace, or specified as a path option using the ``--additional-packs`` argument. When you compile and run queries locally, source dependencies override any dependencies found in the local package cache as well as version constraints defined in the ``qlpack.yml``.
+
+  This is particularly useful in the following situations:
+
+  - One of the dependencies of the query pack you are running is not yet published. Resolving from source is the only way to refernce that pack.
+  - You are making changes to multiple packs at the same time and want to test them together. Resolving from source ensures that you are using the version of the pack with your changes in it.
+
+Example
+~~~~~~~
+
+Consider the following ``codeql-workspace.yml`` file:
+
+.. code-block:: yaml
+
+    provide:
+      - "**/qlpack.yml"
+
+And the following CodeQL library pack ``qlpack.yml`` file in the workspace:
+
+.. code-block:: yaml
+
+    name: my-company/my-library
+    library: true
+    version: 1.0.0
+
+And the following CodeQL query pack ``qlpack.yml`` file in the workspace:
+
+.. code-block:: yaml
+
+    name: my-company/my-queries
+    version: 1.0.0
+    dependencies:
+      my-company/my-library: "*"
+      codeql/cpp-all: ~0.2.0
+
+Notice that the ``dependencies`` block for the CodeQL query pack,``my-company/my-queries``, specifies  ``"*"`` as the version of the library pack. Since the library pack is already defined as a source dependency in ``codeql-workspace.yml``, the library pack's content is always resolved from inside the workspace. Any version constraint you define will be ignored in this case. We recommend that you use ``"*"`` for source dependencies to make it clear that the version is inherited from the workspace.
```

Comment:
Formatting issue in the generated content:

<img width="856" alt="image" src="https://user-images.githubusercontent.com/1877141/188407283-3c855236-9b7e-45ec-9078-12ccf3727572.png">

Likely fix.

```suggestion
Notice that the ``dependencies`` block for the CodeQL query pack, ``my-company/my-queries``, specifies  ``"*"`` as the version of the library pack. Since the library pack is already defined as a source dependency in ``codeql-workspace.yml``, the library pack's content is always resolved from inside the workspace. Any version constraint you define will be ignored in this case. We recommend that you use ``"*"`` for source dependencies to make it clear that the version is inherited from the workspace.
```